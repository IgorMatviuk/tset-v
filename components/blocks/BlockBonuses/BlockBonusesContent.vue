<template>
  <div class="bonuses-content">
    <BContainer>
      <div class="bonuses-content__container" v-html="content"></div>
    </BContainer>
  </div>
</template>

<script>
export default {
  props: {
    content: {
      type: String,
      required: true
    }
  }
}
</script>

<style lang="scss">
.bonuses-content {
  margin-bottom: 30px;

  &__container {
    padding-left: 17px;

    h1,
    h2 {
      position: relative;
      margin-bottom: 31px;
      font-size: 16px;
      font-weight: 400;
      line-height: 20px;
      text-transform: uppercase;
      letter-spacing: 0.05em;

      &::after {
        position: absolute;
        bottom: -16px;
        left: -15px;
        width: 90px;
        height: 1px;
        content: '';
        background-color: $primary-light-1;
      }
    }

    h3,
    h4,
    h5,
    h6 {
      margin-bottom: 16px;
      font-size: 13px;
      font-weight: 500;
      line-height: 16px;
      letter-spacing: 0.05em;
    }

    p {
      margin-bottom: 0;
      font-size: 12px;
      line-height: 15px;
      color: $text-color;

      &:not(:last-child) {
        margin-bottom: 15px;
      }
    }
  }

  @include media-breakpoint-up(md) {
    &__container {
      padding-left: 0;

      h1,
      h2 {
        margin-bottom: 36px;

        &::after {
          bottom: -21px;
          left: -10px;
          width: 100px;
        }
      }
    }
  }

  @include media-breakpoint-up(lg) {
    &__container {
      width: 770px;
      padding-left: 0;
      margin-right: auto;
      margin-left: auto;

      h1,
      h2 {
        &::after {
          left: -30px;
        }
      }
    }
  }

  @include media-breakpoint-up(xl) {
    &__container {
      width: 970px;

      h1,
      h2 {
        font-size: 18px;
        line-height: 23px;

        &::after {
          left: -34px;
          width: 110px;
        }
      }

      h3,
      h4,
      h5,
      h6 {
        margin-bottom: 13px;
        font-size: 14px;
        line-height: 18px;
      }

      p {
        font-size: 13px;
        line-height: 16px;

        &:not(:last-child) {
          margin-bottom: 17px;
        }
      }
    }
  }

  @include media-breakpoint-up(xxl) {
    &__container {
      width: 1170px;
    }
  }
}
</style>
