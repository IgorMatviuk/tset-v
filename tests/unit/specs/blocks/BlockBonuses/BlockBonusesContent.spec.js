import { shallowMount, createLocalVue } from '@vue/test-utils'
import _cloneDeep from 'lodash.clonedeep'

import { BContainer } from 'bootstrap-vue'

import BlockBonusesContent from '@/components/blocks/BlockBonuses/BlockBonusesContent'

const localVue = createLocalVue()
localVue.component('BContainer', BContainer)

describe('BlockBonusesContent', () => {
  let content

  function createBlockBonusesContent() {
    const wrapper = shallowMount(BlockBonusesContent, {
      localVue,
      propsData: {
        content: _cloneDeep(content)
      }
    })

    return wrapper
  }

  beforeEach(() => {
    content = '<p>Some content</p>'
  })

  test('create bonuses content block', () => {
    const wrapper = createBlockBonusesContent()

    expect(wrapper.element).toMatchSnapshot()
  })
})
