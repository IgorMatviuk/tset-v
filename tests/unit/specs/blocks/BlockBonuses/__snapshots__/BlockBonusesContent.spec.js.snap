// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockBonusesContent create bonuses content block 1`] = `
<div
  class="bonuses-content"
>
  <bcontainer-stub
    tag="div"
  >
    <div
      class="bonuses-content__container"
    >
      <p>
        Some content
      </p>
    </div>
  </bcontainer-stub>
</div>
`;
